### PR TITLE
Fix #77. Add a function to delete an imagestream not owned by the component

### LINKED ISSUE #77: imagestream is not deleted when a component - build is removed
## Problem

The imagestream is not deleted when a CRD component - build is removed. This is due to the fact that the imagestream, is created by the openshift controller and don't include the owner reference of our component

We should delete it manually then using as selectror the name of the build image

### PR DESCRIPTION
**Pull request by cmoulliard**
- Add a function to delete an imageStream not owned by the component CRD
- Delete the imageStream if running on OCP
- Use unstructured object to avoid to have to import OpenShift API Type

### CODE REVIEW

**Comment by metacosm** [comment]
This should be part of a `Delete` call that handles clean-up when a component is deleted.

File: pkg/controller/component/others.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+			"apiVersion": "image.openshift.io/v1",
+			"kind":       "ImageStream",
+			"metadata": map[string]interface{}{
+				"name":      request.Name,
+				"namespace": request.Namespace,
+			},
+		},
+	}
+
+	e := r.client.Get(context.TODO(), types.NamespacedName{Namespace: request.Namespace, Name: request.Name}, imageStream)
```
Comment (metacosm) [inline pkg/controller/component/others.go]: Why not do the delete directly instead of fetching it first?

Comment (cmoulliard) [reply pkg/controller/component/others.go]: To be sure that it exists. Maybe this step is not needed after all.

**Comment by cmoulliard** [comment]
LGTM. To be tested when of course it will be possible to use again `buildah and/or kaniko` to perform a docker push to the internal docker registry. Can be merged ;-) @metacosm

**Pull request merged**